### PR TITLE
PHP 5.5 goodness for 3.2

### PR DESCRIPTION
**Pull request by ADmad**
Just looking for confirmation to go ahead and replace all string classes with ::class usage like I have done in `Cache` class.

### CODE REVIEW

**Comment by lorenzo** [comment]
Seems like change for the sake of change... But I think it is fine to star using 5.5 features

File: src/Cache/Cache.php [hunk 1]
```diff
@@ -125,7 +125,7 @@ public static function registry(ObjectRegistry $registry = null)
         if ($registry) {
             static::$_registry = $registry;
         }
-
```
Comment (jippi) [inline src/Cache/Cache.php]: this one is my favorite change - finally we can trim lines in 5.5 !

Comment (ADmad) [inline src/Cache/Cache.php]: @jippi Not my fault if people still haven't discovered the auto blank space trimming feature in their editor/ide.

**Comment by ADmad** [comment]
@lorenzo True, it's was just a cosmetic change. Also I realized I didn't want to do the chore of finding and updating such usages throughout the codebase so i dropped that change :)

Unless someone wants to use this branch to add more changes this is good to merge.

**Comment by markstory** [comment]
This seems like a good place to start. I'm not a fan of change for the sake of change. However, if using `::class` gives us some µs back I'd be all for it.

**Comment by lorenzo** [comment]
Actually, although very minor, this change will make the code slower. The ::class construct is an extra opcode for converting the token into a string after resolving the namespace

**Pull request merged**